### PR TITLE
Get paging properties from parent models

### PR DESCRIPTION
**Pull request by kazrael2119**
fixes https://github.com/Azure/autorest.typescript/issues/2752

### CODE REVIEW

File: packages/autorest.typescript/test/integration/generated/appconfiguration/src/models/index.ts [hunk 1]
```diff
@@ -68,6 +68,20 @@ export interface Label {
...
+  summary?: string;
+  /** The collection value. */
+  items?: Key[];
+  /** The URI that can be used to request the next set of paged results. */
+  nextLink?: string;
+  /** The collection value. */
+  items?: Key[];
+  /** The URI that can be used to request the next set of paged results. */
+  nextLink?: string;
+}
```
Comment (MaryGao) [inline packages/autorest.typescript/test/integration/generated/appconfiguration/src/models/index.ts:83]: After debugging I found this is not js codegen issue but a m4 issue which gives us duplicated nextLink and items properties.
image

Comment (MaryGao) [reply packages/autorest.typescript/test/integration/generated/appconfiguration/src/models/index.ts:83]: @kazrael2119 Could you open an issue in [autorest side](https://github.com/Azure/autorest/issues) and remove swagger case for now?

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 2]
```diff
@@ -435,8 +435,12 @@ function extractPagingType(type: Type, itemName?: string): Type | undefined {
   if (!itemName) {
     return undefined;
   }
-  const prop = (type.properties ?? [])
-    ?.filter((prop) => prop.restApiName === itemName)
+  const allProperties = [
+    ...(type.properties ?? []),
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:439]: Could you add a UT in https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularUnit/operations.spec.ts#L606?

File: packages/autorest.typescript/test/integration/generated/appconfiguration/src/models/index.ts (hunk 1 above)
Comment (kazrael2119) [reply packages/autorest.typescript/test/integration/generated/appconfiguration/src/models/index.ts]: https://github.com/Azure/autorest/issues/5012

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts (hunk 2 above)
Comment (kazrael2119) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts:439]: https://github.com/Azure/autorest.typescript/pull/2764/commits/d4e21d788abff3eba0768e9d93fe0eb26885d093

File: packages/autorest.typescript/src/generators/modelsGenerator.ts [hunk 3]
```diff
@@ -801,6 +801,24 @@ function getNonDiscriminatorProperties(
   return properties.filter(property => !property.isDiscriminator);
 }
 
+/**
+ * Deduplicates properties from an object by removing duplicated ones
+ */
+function deduplicateProperties(
```
Comment (MaryGao) [inline packages/autorest.typescript/src/generators/modelsGenerator.ts:807]: @kazrael2119 pls verify in billing RP.

Comment (kazrael2119) [reply packages/autorest.typescript/src/generators/modelsGenerator.ts:807]: image
this works for billing

**Review by MaryGao — APPROVED** [review]
LGTM

**Pull request merged**